File: 2022/02/0201.py
f = open("input.txt", "r")

# A, X = 1 (rock)
# B, Y = 2 (paper)
# C, Z = 3 (scissors)

# W = 6
# D = 3
# L = 0

x = { 'B': 0, 'A': 3, 'C': 6 }
y = { 'C': 0, 'B': 3, 'A': 6 }
z = { 'A': 0, 'C': 3, 'B': 6 }

def main():
    score = 0;
    for x in f:
        match = x.replace("\n", "").split(" ");
        score += getScore(match)
    print(score);
        

def getScore(match:list[int]) -> int:
    match match[1]:
        case 'X':
            return x[match[0]] + 1;
        case 'Y':
            return y[match[0]] + 2;
        case 'Z':
            return z[match[0]] + 3;
        case _:
            print("No matches wuuuut???");

main();

File: 2022/02/0201.kt
import java.io.File

val x = mapOf("B" to 0, "A" to 3, "C" to 6)
val y = mapOf("C" to 0, "B" to 3, "A" to 6)
val z = mapOf("A" to 0, "C" to 3, "B" to 6)
fun main() {

    var score: Int = 0

    File("input.txt").forEachLine { it ->
        var match = it.replace("\n", "").split(" ").toTypedArray()
        score += getScore(match)
    }
    println(score)
}

fun getScore(match: Array<String>): Int {
    when (match[1]) {
        "X" -> {
            var score = x[match[0]]
            if (score != null) {
                return score + 1
            }
        }
        "Y" -> {
            var score = y[match[0]]
            if (score != null) {
                return score + 2
            }
        }
        "Z" -> {
            var score = z[match[0]]
            if (score != null) {
                return score + 3
            }
        }
        else -> {
            println("wut")
        }
    }
    return -1
}